### PR TITLE
Expand GNNModel arguments and add load function

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #

Changes made in this Pull Request:

 - Adds a `load` function and an example pre-production model
 - Update the model with extra dropout parameters to load the example model

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: openff/nagl/tests/nn/test_model.py [hunk 1]
```diff
@@ -181,7 +179,14 @@ def test_compute_property_networkx(self, am1bcc_model, openff_methane_uncharged)
...
         expected = np.array([-0.143774, 0.035943, 0.035943, 0.035943, 0.035943])
         assert_allclose(charges, expected, atol=1e-5)
+
+    def test_load(self, openff_methane_uncharged):
+        model = GNNModel.load("openff-gnn-am1bcc-0.0.1-alpha.1.pt")
+        assert isinstance(model, GNNModel)
+
+        charges = model.compute_property(openff_methane_uncharged, as_numpy=True)
+        expected = np.array([-0.111393,  0.027848,  0.027848,  0.027848,  0.027848])
+        assert_allclose(charges, expected, atol=1e-5)
```
Comment (lilyminium) [inline openff/nagl/tests/nn/test_model.py:192]: Based on a meeting with @mattwthompson and @j-wags, we decided not to ship models with NAGL (as is demonstrated in this PR) but to have a separate data repo. There was also some discussion about not using `torch.load` as that just unpickles the model. I was going to look into `load_from_checkpoint` to see if that was any safer, or consider packaging a NAGL-specific format.

Comment (lilyminium) [reply openff/nagl/tests/nn/test_model.py]: I switched formats to save a dictionary of model parameters and weights; this at least is more future-proof against updating the classes, although it doesn't solve the pickle problem.

**Review by mattwthompson — APPROVED** [review]
This is nitpicky but it'd be nice to have `test` somewhere in the path to the example model. In terms of how it's used I don't think there's any confusion about whether or not it's a released model. But a user passing by might reasonably think it's something that's shipped; the path `openff/nagl/data/example_am1bcc_model.pt` doesn't necessarily communicate that it's for testing purposes only. Not blocking / feel free to decline this suggestion, I have a slight preference for tucking the data directory behind `openff/foobar/tests/data` or something like that.

Otherwise this is wonderful and I'm super excited to get to use it!

```python

In [1]: from openff.nagl.nn._models import GNNModel
/Users/mattthompson/mambaforge/envs/nagl/lib/python3.10/site-packages/lightning_utilities/core/imports.py:13: UserWarning: Module openff was already imported from None, but /Users/mattthompson/software/openff-nagl is being added to sys.path
  import pkg_resources

In [2]: model = GNNModel.load("openff/nagl/data/example_am1bcc_model.pt")

In [3]: from openff.toolkit import Molecule

In [4]: molecule = Molecule.from_smiles("c1ccc2c(c1)C=Cc3ccccc3N2C(=O)N")

In [5]: %%timeit
   ...: molecule.assign_partial_charges(partial_charge_method="am1bcc")
   ...:
   ...:
6.31 s ± 316 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [6]: original_charges = molecule.partial_charges

In [7]: %%timeit
   ...: model.compute_property(molecule)
   ...:
   ...:
/Users/mattthompson/software/openff-nagl/openff/nagl/molecule/_graph/_graph.py:165: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  data = {k: torch.tensor(v[edge_indices.long()]) for k, v in self.edata.items()}
121 ms ± 8.73 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [8]: molecule.partial_charges - original_charges
Out[8]:
array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]) <Unit('elementary_charge')>

File: openff/nagl/data/files.py [hunk 2]
```diff
@@ -22,3 +22,4 @@
 EXAMPLE_MODEL_CONFIG = resource_filename(__name__, "example_model_config.yaml")
 MODEL_CONFIG_V7 = resource_filename(__name__, "model_config_v7.yaml")
 EXAMPLE_AM1BCC_MODEL_STATE_DICT = resource_filename(__name__, "example_am1bcc_model_state_dict.pt")
+EXAMPLE_AM1BCC_MODEL = resource_filename(__name__, "example_am1bcc_model.pt")
```
Comment (mattwthompson) [inline openff/nagl/data/files.py:25]: Slick little trick, I tend to forget the paths/names of my test files so I'm going to steal this

**Comment by lilyminium** [comment]
Thanks for the review, @mattwthompson! I've moved the data directory so the path is now `GNNModel.load("openff/nagl/tests/data/example_am1bcc_model.pt")`. Once this is merged I'll also open a PR fixing those warnings you're seeing.
